### PR TITLE
#31 Protobuf

### LINKED ISSUE #31: Serialize data with protobuf
we want to serialize the data were sending using protobuf. Convert the current json decoding in data.rs to be in protobuf instead

useful resources
https://docs.rs/protobuf/latest/protobuf/

chatgpt

### PR DESCRIPTION
**Pull request by Peyton-McKee**
Implemented Protobuf and added documentation for generating code

closes #31

### CODE REVIEW

File: src/serverdata.rs [hunk 1]
```diff
@@ -0,0 +1,209 @@
+// This file is generated by rust-protobuf 3.3.0. Do not edit
+// .proto file is parsed by protoc --rust-out=...
```
Comment (jr1221) [inline src/serverdata.rs:2]: Should prob ignore this?

**Comment by jr1221** [comment]
This is 100% not something needed and definitely overkill but we can put a [cargo build script in](https://doc.rust-lang.org/cargo/reference/build-scripts.html) which could re-run the protoc command [if the .proto file changes.](https://doc.rust-lang.org/cargo/reference/build-scripts.html#rerun-if-changed)

File: src/serverdata.rs (hunk 1 above)
Comment (Peyton-McKee) [reply src/serverdata.rs:2]: I dont think itll compile in github for our cd checks

Comment (jr1221) [reply src/serverdata.rs:2]: ah ok nvm then.

**Comment by nwdepatie** [comment]
> This is 100% not something needed and definitely overkill but we can put a [cargo build script in](https://doc.rust-lang.org/cargo/reference/build-scripts.html) which could re-run the protoc command [if the .proto file changes.](https://doc.rust-lang.org/cargo/reference/build-scripts.html#rerun-if-changed)

Note that we are gonna need a build script for Rust generation in the near future too

**Comment by Peyton-McKee** [comment]
> > This is 100% not something needed and definitely overkill but we can put a [cargo build script in](https://doc.rust-lang.org/cargo/reference/build-scripts.html) which could re-run the protoc command [if the .proto file changes.](https://doc.rust-lang.org/cargo/reference/build-scripts.html#rerun-if-changed)
> 
> Note that we are gonna need a build script for Rust generation in the near future too

yeah we can just add this to that ticket ig, i dont want to deal with that rn

**Pull request merged**